cpp/include/patterns/transform_reduce_e.cuh

```diff
@@ -0,0 +1,220 @@
+                            ? matrix_partition.get_major_from_major_offset_nocheck(idx)
+                            : minor;
+      auto row_offset =
+        GraphViewType::is_adj_matrix_transposed ? minor_offset : static_cast<vertex_t>(idx);
+      auto col_offset =
+        GraphViewType::is_adj_matrix_transposed ? static_cast<vertex_t>(idx) : minor_offset;
+      auto e_op_result = evaluate_edge_op<GraphViewType,
+                                          AdjMatrixRowValueInputIterator,
+                                          AdjMatrixColValueInputIterator,
+                                          EdgeOp>()
+                           .compute(row,
+                                    col,
+                                    weight,
+                                    *(adj_matrix_row_value_input_first + row_offset),
+                                    *(adj_matrix_col_value_input_first + col_offset),
+                                    e_op);
+      e_op_result_sum = plus_edge_op_result(e_op_result_sum, e_op_result);
+    }
+    idx += gridDim.x * blockDim.x;
+  }
+
+  e_op_result_sum =
+    block_reduce_edge_op_result<e_op_result_t,
+                                transform_reduce_e_for_all_low_out_degree_block_size>()
+      .compute(e_op_result_sum);
+  if (threadIdx.x == 0) { *(block_result_first + blockIdx.x) = e_op_result_sum; }
+}
+
+}  // namespace detail
+
+/**
+ * @brief Iterate over the entire set of edges and reduce @p edge_op outputs.
+ *
+ * This function is inspired by thrust::transform_reduce().
+ *
+ * @tparam GraphViewType Type of the passed non-owning graph object.
+ * @tparam AdjMatrixRowValueInputIterator Type of the iterator for graph adjacency matrix row
+ * input properties.
+ * @tparam AdjMatrixColValueInputIterator Type of the iterator for graph adjacency matrix column
+ * input properties.
+ * @tparam EdgeOp Type of the binary (or ternary) edge operator.
+ * @tparam T Type of the initial value.
+ * @param handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator, and
+ * handles to various CUDA libraries) to run graph algorithms.
+ * @param graph_view Non-owning graph object.
+ * @param adj_matrix_row_value_input_first Iterator pointing to the adjacency matrix row input
+ * properties for the first (inclusive) row (assigned to this process in multi-GPU).
+ * `adj_matrix_row_value_input_last` (exclusive) is deduced as @p adj_matrix_row_value_input_first +
+ * @p graph_view.get_number_of_adj_matrix_local_rows().
+ * @param adj_matrix_col_value_input_first Iterator pointing to the adjacency matrix column input
+ * properties for the first (inclusive) column (assigned to this process in multi-GPU).
+ * `adj_matrix_col_value_output_last` (exclusive) is deduced as @p adj_matrix_col_value_output_first
+ * + @p graph_view.get_number_of_adj_matrix_local_cols().
+ * @param e_op Binary (or ternary) operator takes *(@p adj_matrix_row_value_input_first + i), *(@p
```

Comment:
Documentation indicates this should be a binary or ternary operator.  However, it requires 4 or 5 parameters:
`(vertex_t src, vertex_t dst, weight_t w, auto src_val, auto dst_val)`
or (presumably)
`(vertex_t src, vertex_t dst, auto src_val, auto dst_val)`

Reply:
Yes, this comment needs to be fixed. Thanks for catching this.

Reply:
Fixed.